### PR TITLE
[docs] Added java.md to developer docs

### PR DESCRIPTION
**Pull request by FelixGV**
Other minor changes:

- Added instructions to avoid double brace init in the sytle guide.

- Tweaked learn_more.md to include additional podcast links.

- Added several SVG icons under the assets directory and sprinkled them over a few pages.

## How was this PR tested?
Doc changes can be seen on my fork. The following links have changed: 

- https://felixgv.github.io/venice/
- https://felixgv.github.io/venice/docs/dev_guide/java
- https://felixgv.github.io/venice/docs/dev_guide/how_to/style_guide

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: docs/dev_guide/how_to/style_guide.md [hunk 1]
```diff
@@ -131,6 +140,75 @@ types (e.g., `-1` for a numeric value that is otherwise expected to be positive)
...
+
+There is an antipattern which consists of initializing a collection (such as a map) with "double braces" in order to add
+elements into it. This is a combination of the above two techniques, with the first (outer) set of braces denoting the
+anonymous class, and the second (inner) set of braces denoting a static block. For example:
+
+```java
+class Sample3 {
+  void sample() {
+    // antipattern:
+    Map<String, String> map1 = new HashMap<>() {
```
Comment (majisourav99) [inline docs/dev_guide/how_to/style_guide.md:187]: I saw/used this pattern in tests, but never realized the hidden cost of anonymous classes. 👍

**Review by majisourav99 — APPROVED** [review]
Thanks so much for helpful docs!

**Review by majisourav99 — APPROVED** [review]
Thanks for including the node.js reference :)

**Comment by FelixGV** [comment]
Thanks for the review @majisourav99 !

**Pull request merged**